posthog/api/test/test_action.py

```diff
@@ -124,18 +166,46 @@ def test_update_action(self, patch_capture, *args):
             },
             HTTP_ORIGIN="http://testserver",
         )
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(response.json()["name"], "user signed up 2")
-        self.assertEqual(response.json()["created_by"], None)
-        self.assertEqual(response.json()["steps"][0]["id"], str(action_id))
-        self.assertEqual(response.json()["steps"][1]["href"], "/a-new-link")
-        self.assertEqual(response.json()["description"], "updated description")
+        assert response.status_code == status.HTTP_200_OK, response.json()
+        assert response.json()["name"] == "user signed up 2"
+        assert response.json()["description"] == "updated description"
+        assert not response.json()["created_by"]
+        assert response.json()["steps"] == [
+            {
+                "event": None,
+                "properties": [{"key": "$browser", "value": "Chrome"}],
+                "selector": "div > button",
+                "tag_name": None,
+                "text": "sign up NOW",
+                "text_matching": None,
+                "href": None,
+                "href_matching": None,
+                "url": None,
+                "url_matching": "contains",
+            },
+            {
+                "event": None,
+                "properties": None,
+                "selector": None,
+                "tag_name": None,
+                "text": None,
+                "text_matching": None,
+                "href": "/a-new-link",
+                "href_matching": None,
+                "url": None,
+                "url_matching": "contains",
+            },
+        ]
 
         action.refresh_from_db()
-        steps = action.steps.all().order_by("id")
-        self.assertEqual(action.name, "user signed up 2")
-        self.assertEqual(steps[0].text, "sign up NOW")
-        self.assertEqual(steps[1].href, "/a-new-link")
+        assert action.name == "user signed up 2"
+
+        assert action.action_steps.count() == 2
+        assert action.action_steps.all()[0].text == "sign up NOW"
+        assert action.action_steps.all()[1].href == "/a-new-link"
+
+        # TODO: This can't be right - For @marius to look into perhaps?
+        assert action.bytecode == ["_h", 29, 32, "Chrome", 32, "$browser", 32, "properties", 1, 2, 11, 4, 2]
```

Comment:
Not sure if I fully see what's wrong, but @mariusandra

Reply:
Meant for @mariusandra - to my eye that bytecode can't possibly be enough for the action to be matched with its config. Wasn't sure that is well understood or if there is something I missed

Reply:
Figured it out 🤦 - the steps were missing "event" which is used by the generator to decide what bytecode needs to be there